bin/p2-dsctl/main.go

```diff
@@ -0,0 +1,237 @@
+
+	cmdList = kingpin.Command(CmdList, "List daemon sets.")
+
+	cmdEnable = kingpin.Command(CmdEnable, "Enable daemon set.")
+	enableID  = cmdEnable.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdDisable = kingpin.Command(CmdDisable, "Disable daemon set.")
+	disableID  = cmdDisable.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdDelete = kingpin.Command(CmdDelete, "Delete daemon set.")
+	deleteID  = cmdDelete.Arg("id", "The uuid for the daemon set").Required().String()
+
+	cmdUpdate       = kingpin.Command(CmdUpdate, "Update a daemon set.")
+	updateID        = cmdUpdate.Arg("id", "The uuid for the daemon set").Required().String()
+	updateAZ        = cmdUpdate.Flag("az", "The availability zone of the pod cluster").String()
+	updateManifest  = cmdUpdate.Flag("manifest", "Complete manifest - must parse as JSON!").String()
+	updateMinHealth = cmdUpdate.Flag("minhealth", "The minimum health of the daemon set").String()
+	updateName      = cmdUpdate.Flag("name", "The cluster name (ie. staging, production)").String()
+	updatePodID     = cmdUpdate.Flag("pod", "The pod ID on the daemon set").String()
+)
+
+func main() {
+	cmd, consulOpts := flags.ParseWithConsulOptions()
+	client := kp.NewConsulClient(consulOpts)
+	logger := logging.NewLogger(logrus.Fields{})
+	dsstore := dsstore.NewConsul(client, 3, &logger)
+
+	switch cmd {
+	case CmdCreate:
+		az := pc_fields.AvailabilityZone(*createAZ)
+		minHealth, err := strconv.Atoi(*createMinHealth)
+		if err != nil {
+			log.Fatalf("Invalid value for minimum health, expected integer: %v", err)
+		}
+		name := ds_fields.ClusterName(*createName)
+		podID := types.PodID(*createPodID)
+		selector := selectorFrom(az)
+
+		manifest, err := pods.ManifestFromPath(*createManifest)
+		if err != nil {
+			log.Fatalf("%s", err)
+		}
+
+		ds, err := dsstore.Create(manifest, minHealth, name, selector, podID)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("%v has been created in consul", ds.ID)
+		fmt.Println()
+
+	case CmdGet:
+		id := ds_fields.ID(*getID)
+		ds, _, err := dsstore.Get(id)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		bytes, err := json.Marshal(ds)
+		if err != nil {
+			logger.WithError(err).Fatalln("Unable to marshal daemon set as JSON")
+		}
+		fmt.Printf("%s", bytes)
+
+	case CmdList:
+		dsList, err := dsstore.List()
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		for _, ds := range dsList {
+			fmt.Println(ds.ID)
+		}
+
+	case CmdEnable:
+		id := ds_fields.ID(*enableID)
+
+		mutator := func(ds ds_fields.DaemonSet) (ds_fields.DaemonSet, error) {
+			if !ds.Disabled {
+				return ds, util.Errorf("Daemon set has already been enabled")
+			}
+			ds.Disabled = false
+			return ds, nil
+		}
+
+		_, err := dsstore.MutateDS(id, mutator)
+		if err != nil {
+			log.Fatalf("err: %v", err)
+		}
+		fmt.Printf("The daemon set '%s' has been successfully enabled in consul", id.String())
+		fmt.Println()
```

Comment:
do we like this explicit `Println` better than appending a `\n` to previous message?

I don't care, but curious